src/H5Centry.c

```diff
@@ -885,12 +881,6 @@ H5C__flush_single_entry(H5F_t *f, H5C_cache_entry_t *entry_ptr, unsigned flags)
                                 "can't notify client about entry dirty flag cleared")
             } /* end if */
 
-            /* we are about to discard the in core representation --
-             * set the magic field to bad magic so we can detect a
-             * freed entry if we see one.
-             */
-            entry_ptr->magic = H5C__H5C_CACHE_ENTRY_T_BAD_MAGIC;
-
```

Comment:
I feel like we need to expunge the term "in core" from the library. "Core" stopped being a common synonym for "memory" a long time ago.